### PR TITLE
[improvement](log)Support exception report in our benchmark utils (#40834)

### LINKED ISSUE #40834: [Enhancement](good first issue) Support exception report in our benchmark utils
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

now some of our benchmark utils(see in `tools/`, like clickbench, tpch...) don't report detail messages when meet error running query. it's not very convinced. we need support this.

### Solution

modify the scripts to report error if meet.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by cat-with-cat**
## Proposed changes

Issue Number: close #40834 

```
➜  doris_ bash /home/cat/doris_/doris/tools/clickbench-tools/run-clickbench-queries.sh
query1,ERROR 1105 (HY000) at line 1: errCode = 2, detailMessage = There is no scanNode Backend available.[10004: not alive]
,ERROR 1105 (HY000) at line 1: errCode = 2, detailMessage = There is no scanNode Backend available.[10004: not alive]
,ERROR 1105 (HY000) at line 1: errCode = 2, detailMessage = There is no scanNode Backend available.[10004: not alive]

```

```
➜  doris_ bash /home/cat/doris_/doris/tools/ssb-tools/bin/run-ssb-flat-queries.sh
============================================
q1.1    317     112     107     107
q1.2    91      52      52      52
q1.3    198     101     136     101
q2.1    3197    1751    1642    1642
q2.2    Error: Failed to execute query q2.2 (cold run). Output:
ERROR 1105 (HY000) at line 17: RpcException, msg: send fragments failed. io.grpc.StatusRuntimeException: UNAVAILABLE: io exception, host: 192.168.41.145
q2.3    Error: Failed to execute query q2.3 (cold run). Output:
ERROR 1105 (HY000) at line 17: RpcException, msg: send fragments failed. io.grpc.StatusRuntimeException: UNAVAILABLE: io exception, host: 192.168.41.145
```

```
➜  doris_ bash /home/cat/doris_/doris/tools/ssb-tools/bin/run-ssb-queries.sh
============================================
q1.1    881     210     231     210
q1.2    1075    153     176     153
q1.3    131     124     143     124
q2.1    5429    1551    1638    1551
q2.2    1785    1618    1615    1615
q2.3    1234    1189    1418    1189
q3.1    Error: Failed to execute query q3.1 (cold run). Output:
ERROR 1105 (HY000) at line 17: RpcException, msg: send fragments failed. io.grpc.StatusRuntimeException: UNAVAILABLE: io exception, host: 192.168.41.145
q3.2    Error: Failed to execute query q3.2 (cold run). Output:
ERROR 1105 (HY000) at line 17: RpcException, msg: send fragments failed. io.grpc.StatusRuntimeException: UNAVAILABLE: io exception, host: 192.168.41.145
```

```
➜  doris_ bash /home/cat/doris_/doris/tools/clickbench-tools/load-clickbench-data.sh
FE_HOST: 127.0.0.1
FE_HTTP_PORT: 8030
USER: root
PASSWORD: 
DB: clickbench
start...
Curl failed with the following output:

Error: Failed to execute curl to fetch BE's configuration.
```

```
➜  doris_ bash /home/cat/doris_/doris/tools/clickbench-tools/run-clickbench-queries.sh
mysql  Ver 8.0.39-0ubuntu0.22.04.1 for Linux on x86_64 ((Ubuntu))

This is perl 5, version 34, subversion 0 (v5.34.0) built for x86_64-linux-gnu-thread-multi
(with 60 registered patches, see perl -V for more detail)

Copyright 1987-2021, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at http://www.perl.org/, the Perl Home Page.

FE_HOST: 127.0.0.1
FE_QUERY_PORT: 9030
USER: root
PASSWORD: 
DB: clickbench
ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1:9030' (111)
```

### CODE REVIEW

**Comment by cat-with-cat** [comment]
run buildall

**Review by cat-with-cat — COMMENTED** [review]
Issue Number: #40834
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for fe-common 1.2-SNAPSHOT:
[INFO] 
[INFO] fe-common .......................................... SUCCESS [02:48 min]
[INFO] spark-dpp .......................................... SUCCESS [ 59.651 s]
[INFO] fe-core ............................................ SUCCESS [13:11 min]
[INFO] hive-udf ........................................... SUCCESS [ 28.343 s]
[INFO] java-common ........................................ SUCCESS [  2.664 s]
[INFO] hudi-scanner ....................................... SUCCESS [ 44.719 s]
[INFO] java-udf ........................................... SUCCESS [ 26.569 s]
[INFO] jdbc-scanner ....................................... SUCCESS [ 27.037 s]
[INFO] paimon-scanner ..................................... SUCCESS [ 34.518 s]
[INFO] max-compute-scanner ................................ SUCCESS [04:56 min]
[INFO] avro-scanner ....................................... SUCCESS [ 31.400 s]
[INFO] lakesoul-scanner ................................... SUCCESS [ 21.580 s]
[INFO] preload-extensions ................................. SUCCESS [03:54 min]
[INFO] trino-connector-scanner ............................ SUCCESS [ 28.898 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  16:00 min (Wall Clock)
[INFO] Finished at: 2024-10-30T22:05:10+08:00
[INFO] ------------------------------------------------------------------------
OUTPUT DIR=/home/cat/doris_/doris/output/
***************************************
Successfully build Doris
***************************************
```

**Review by zclllyybb — COMMENTED** [review]
basically looks good. need some more result and shell format

File: tools/ssb-tools/bin/run-ssb-queries.sh [hunk 1]
```diff
@@ -120,19 +123,31 @@ for i in '1.1' '1.2' '1.3' '2.1' '2.2' '2.3' '3.1' '3.2' '3.3' '3.4' '4.1' '4.2'
     hot2=0
     echo -ne "q${i}\t" | tee -a result.csv
     start=$(date +%s%3N)
-    mysql -h"${FE_HOST}" -u "${USER}" -P"${FE_QUERY_PORT}" -D"${DB}" --comments <"${QUERIES_DIR}"/q"${i}".sql >"${RESULT_DIR}"/result"${i}".out 2>"${RESULT_DIR}"/result"${i}".log
+    if ! mysql -h"${FE_HOST}" -u"${USER}" -P"${FE_QUERY_PORT}" -D"${DB}" --comments \
+        <"${QUERIES_DIR}/q${i}.sql" >"${RESULT_DIR}/result${i}.out" 2>"${RESULT_DIR}/result${i}.log"; then
```
Comment (zclllyybb) [inline tools/ssb-tools/bin/run-ssb-queries.sh:127]: maybe don't use the log file, just printing the errmsg to console is better

File: result.csv
Comment (zclllyybb) [inline result.csv:1]: dont upload this file

File: tools/ssb-tools/bin/ssb-dbgen-linux.tar.gz
Comment (zclllyybb) [inline tools/ssb-tools/bin/ssb-dbgen-linux.tar.gz:1]: dont upload this file

**Review by zclllyybb — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**